lib/IDE/CodeCompletionResult.cpp

```diff
@@ -399,9 +399,6 @@ ContextFreeCodeCompletionResult::calculateContextualNotRecommendedReason(
   if (explicitReason != ContextualNotRecommendedReason::None) {
     return explicitReason;
   }
-  if (IsAsync && !canCurrDeclContextHandleAsync) {
-    return ContextualNotRecommendedReason::InvalidAsyncContext;
-  }
```

Comment:
Unless I’m missing something, we can remove `IsAsync` from `ContextFreeCodeCompletionResult` now.